src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
+// eats grass
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
+// eats dudes
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        delete Animal.alive[Animal.alive.indexOf(target)];
```

Comment:
Using `delete` on an array leaves an undefined hole in the array. Instead, you should use the `filter` method to remove the dead animal from the `Animal.alive` array. This avoids the performance issue of nesting loops and adheres to the checklist item regarding the use of `filter` over `indexOf` and `splice`.